applications: serial_lte_modem: Add Linux PPP+CMUX support (#21135)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/PPP_linux.rst

* ldattach from the util-linux package

These should be available on all standard Linux distributions.

Configuration
=============

For building the SLM application use :file:`overlay-ppp-cmux-linux.conf` configuration overlay.

Serial port baud rate may be adjusted from device tree overlay file. By default, the baud rate is set to 115200.
If baud rate is changed, the same baud rate should be set in the :file:`scripts/slm_start_ppp.sh` and :file:`scripts/slm_stop_ppp.sh` scripts.

Flashing and running
====================

Build and flash the SLM application to the nRF91 Series using the overlay file :file:`overlay-ppp-cmux-linux.conf`.

Managing the connection
=======================

There are start and stop scripts provided in the :file:`scripts` directory of the SLM application.
These scripts assume that the nRF91 Series SiP is connected to the Linux device using serial port `/dev/ttyACM0`.

Adjust the serial port settings in the scripts if needed:

.. code-block:: console
   MODEM=/dev/ttyACM0
   BAUD=115200

To start the PPP connection, run the :file:`scripts/slm_start_ppp.sh` script.
To stop the PPP connection, run the :file:`scripts/slm_stop_ppp.sh` script.

> Scripts need superuser privileges to run, so use `sudo` to run them.
> 
> Example usage:

Reviewer: ```suggestion
To run the scripts, you need super user privileges, so use `sudo` as follows:
```

.. _slm_as_linux_modem:

nRF91 Series SiP as a modem for Linux device
############################################

.. contents::
   :local:
   :depth: 2

Overview
********

You can use the Serial LTE Modem (SLM) application to make an nRF91 Series SiP work as a standalone modem that can be used with a Linux device.
The Linux device can use a standard PPP daemon and ldattach utility to connect to the cellular network through the nRF91 Series SiP.

> The setup differentiates from typical dial-up modem connection as we are using GSM 0710 multiplexer protocol (CMUX) to multiplex multiple data streams over a single serial port.
> This allows us to use the same serial port for both AT commands and PPP data.

Reviewer: ```suggestion
The setup differentiates from a typical dial-up modem connection as the GSM 0710 multiplexer protocol (CMUX) is used to multiplex multiple data streams over a single serial port.
This allows you to use the same serial port for both AT commands and PPP data.
```

Prerequisites
=============

The Linux device needs to have the following packages installed:

* pppd from the ppp package
* ldattach from the util-linux package

These should be available on all standard Linux distributions.

Configuration
=============

To build the SLM application, use the :file:`overlay-ppp-cmux-linux.conf` configuration overlay.

You can adjust the serial port baud rate using the devicetree overlay file.
By default, the baud rate is set to 115200.
If you change the baud rate, set the same rate in the :file:`scripts/slm_start_ppp.sh` and :file:`scripts/slm_stop_ppp.sh` scripts.

Building and running
====================

> To build and program the SLM application to the nRF91 Series device, use the  :file:`overlay-ppp-cmux-linux.conf` overlay file.

Reviewer: ```suggestion
To build and program the SLM application to the nRF91 Series device, use the :file:`overlay-ppp-cmux-linux.conf` overlay file.
```

Managing the connection
=======================

The start and stop scripts are provided in the :file:`scripts` directory of the SLM application.
The scripts assume that the nRF91 Series SiP is connected to the Linux device using the `/dev/ttyACM0` serial port.

If needed, adjust the serial port settings in the scripts as follows:

.. code-block:: none

   MODEM=/dev/ttyACM0
   BAUD=115200

To start the PPP connection, run the :file:`scripts/slm_start_ppp.sh` script.
To stop the PPP connection, run the :file:`scripts/slm_stop_ppp.sh` script.

> Scripts need superuser privileges to run, so use `sudo` to run them.

Reviewer: ```suggestion
The scripts need superuser privileges to run, so use `sudo`.
```

The PPP link is set as a default route if there is no existing default route.

> These scripts do not manage DNS settings from Linux system. Use your distributions manuals to study how to configure DNS settings.

Reviewer: ```suggestion
The scripts do not manage the DNS settings from the Linux system.
Read the distribution manuals to learn how to configure the DNS settings.
```

> The following example shows how to start the connection and verify its operation with various command line utilities:

Reviewer: ```suggestion
The following example shows how to start the connection and verify its operation with various command-line utilities:
```
